assets/product-info.js

```diff
@@ -54,21 +53,16 @@ if (!customElements.get('product-info')) {
       fetch("/cart?section_id=main-cart-items")
       .then((response) => response.text())
       .then((responseText) => {
-        const html = new DOMParser().parseFromString(responseText, 'text/html')
-        const sourceQty = html.querySelector((`[data-quantity-variant-id~="${this.currentVariant.value}"]`))
-        const quantityRulesCartClassname = '.quantity__rules-cart';
-        if (sourceQty) {
-          const valueQtyCart = sourceQty.value;         
-          this.querySelector(quantityRulesCartClassname).classList.toggle('hidden', valueQtyCart <= 0);
-          if (valueQtyCart && this.input) {
-            this.input.dataset.cartQuantity = valueQtyCart;
-            this.destinationQty.innerHTML = valueQtyCart;
-          }
-        } else {
-          if (this.input) this.input.dataset.cartQuantity = 0;
-          this.destinationQty.innerHTML = 0;
-          this.querySelector(quantityRulesCartClassname).classList.add('hidden')
+        const html = new DOMParser().parseFromString(responseText, 'text/html');
+        const source = html.querySelector((`[data-quantity-variant-id~="${this.currentVariant.value}"]`));
+        
+        if (source) {
```

Comment:
- Simplified the logic
- Use `querySelector()` to get updated element instead of using cached element, because variant change now replaces the input in the DOM